lms/djangoapps/support/tests/test_tasks.py

```diff
@@ -0,0 +1,99 @@
+from lms.djangoapps.courseware.models import StudentModule
+from lms.djangoapps.support.tasks import reset_student_course
+from lms.djangoapps.support.tests.factories import CourseResetAuditFactory, CourseResetCourseOptInFactory
+from lms.djangoapps.support.models import CourseResetAudit
+from common.djangoapps.student.models.course_enrollment import CourseEnrollment
+from common.djangoapps.student.roles import SupportStaffRole
+from common.djangoapps.student.tests.factories import UserFactory
+
+
+class ResetStudentCourse(TestSubmittingProblems):
+    """ Test expire_waiting_enrollments task """
+    USERNAME = "support"
+    EMAIL = "support@example.com"
+    PASSWORD = "support"
+
+    def setUp(self):
+        """
+        Set permissions, create a course and learner, enroll learner and opt into course reset
+        """
+        super().setUp()
+        self.user = UserFactory(username=self.USERNAME, email=self.EMAIL, password=self.PASSWORD)
+        SupportStaffRole().add_users(self.user)
+        self.course_id = str(self.course.id)
+        self.enrollment = CourseEnrollment.objects.filter(user=self.student_user, course_id=self.course.id).first()
+        self.opt_in = CourseResetCourseOptInFactory.create(course_id=self.course.id)
+        self.audit = CourseResetAuditFactory.create(
+            course=self.opt_in,
+            course_enrollment=self.enrollment,
+            reset_by=self.user,
+            status=CourseResetAudit.CourseResetStatus.FAILED
+        )
+
+    def basic_setup(self, late=False, reset=False, showanswer=False):
+        """
+        Set up a simple course for testing basic grading functionality.
+        """
+        grading_policy = {
+            "GRADER": [{
+                "type": "Homework",
+                "min_count": 1,
+                "drop_count": 0,
+                "short_label": "HW",
+                "weight": 1.0
+            }],
+            "GRADE_CUTOFFS": {
+                'A': .9,
+                'B': .33
+            }
+        }
+        self.add_grading_policy(grading_policy)
+
+        # set up a simple course with four problems
+        homework = self.add_graded_section_to_course('homework', late=late, reset=reset, showanswer=showanswer)
+        vertical = BlockFactory.create(
+            parent_location=homework.location,
+            category='vertical',
+            display_name='Subsection 1',
+        )
+        self.add_dropdown_to_section(vertical.location, 'p1', 1)
+        self.add_dropdown_to_section(vertical.location, 'p2', 1)
+        self.add_dropdown_to_section(vertical.location, 'p3', 1)
+
+        self.refresh_course()
+
+    def test_reset_student_course(self):
+        self.basic_setup()
+        reset_student_course(self.course_id, self.student_user.email, self.user.email)
+        course_reset_audit = CourseResetAudit.objects.filter(course_enrollment=self.enrollment).first()
```

Comment:
```suggestion
        course_reset_audit = CourseResetAudit.objects.get(course_enrollment=self.enrollment)
```
We only expect there to be one, so this is cleaner, and hey now, the test will fail if there's somehow more than one audit for the enrollment